supabase/functions/process-audio/index.ts

```diff
@@ -1,8 +1,49 @@
 import { serve } from "https://deno.land/std@0.170.0/http/server.ts";
 import OpenAI, { toFile } from "https://deno.land/x/openai@v4.26.0/mod.ts";
-import { multiParser } from 'https://deno.land/x/multiparser@0.114.0/mod.ts';
+import { decodeBase64 } from "https://deno.land/std@0.217.0/encoding/base64.ts";
 import { corsHeaders } from "../common/cors.ts";
 import { supabaseClient } from "../common/supabaseClient.ts";
+import { multiParser } from 'https://deno.land/x/multiparser@0.114.0/mod.ts';
+
+
+function createWavHeader(dataLength: number, sampleRate: number, numChannels: number, bitDepth: number) {
```

Comment:
I guess this is related to the the issue you described of not being able to send the raw bytes?

Are you able to see / log the raw bytes locally on the phone? 

Meaning, can you work with the bytes locally, just can't send them, or you can't even collect them as raw bytes?

Reply:
I get raw bytes from esp32, so I can get them as raw bytes and populate an `ArrayBuffer`, but sending just the array buffer over HTTP Capacitor client did not work, as in no request was made. So instead I encoded that array buffer as base64 and sent it as part of a JSON.

Reply:
I see - so the header can still be managed by the client side, and just base64 the entire bytes of the wav file?

To make the breaking change smaller, or even not exists

You can add an if statement that checks if it is base64, do x else y

Reply:
I added several if statements, so now it no longer is a breaking change, it will still work with form requests and binary bodies.
```

    if (contentType.includes('multipart/form-data')) {
        const form = await multiParser(req);
        if (!form || !form.files || !form.files.file) {
            return new Response('File not found in form', {
                status: 400,
                headers: corsHeaders,
            });
        }
        console.log('Form:', form);
        const file = form.files.file;
        arrayBuffer = file.content.buffer;
        filenameTimestamp = file.filename || filenameTimestamp;
    } else if (contentType.includes('application/json')) {
      const { data } = await req.json();
      const audioData = decodeBase64(data);

      console.log('Audio data:', audioData.length);
      // 1 Channel, 8000 sample rate, 16 bit depth
      const wavHeader = createWavHeader(audioData.length, 16000, 1, 16);
      const wavBytes = new Uint8Array(wavHeader.length + audioData.length);
      wavBytes.set(wavHeader, 0);
      wavBytes.set(audioData, wavHeader.length);
      arrayBuffer = wavBytes.buffer;
    } else {
        arrayBuffer = await req.arrayBuffer();
    }
    ```